cmd/rehearse/main.go

```diff
@@ -226,25 +227,62 @@ func execute(jobs config.JobConfig, jobSpec *pjapi.ProwJobSpec, logger logrus.Fi
 		}
 	}
 
-	if len(rehearsals) > 0 {
-		if err := rehearsalConfigs.Create(); err != nil {
-			return fmt.Errorf("failed to prepare rehearsal ci-operator config ConfigMap: %v", err)
-		}
-		for _, job := range rehearsals {
-			created, err := submitRehearsal(job, jobSpec, logger, pjclient, dry)
-			if err != nil {
-				logger.WithError(err).Warn("Failed to execute a rehearsal presubmit presubmit")
-			} else {
-				logger.WithFields(pjutil.ProwJobFields(created)).Info("Submitted rehearsal prowjob")
-			}
-		}
-	} else {
+	if len(rehearsals) == 0 {
 		logger.Warn("No job rehearsals")
+		return nil
+	}
+	if err := rehearsalConfigs.Create(); err != nil {
+		return fmt.Errorf("failed to prepare rehearsal ci-operator config ConfigMap: %v", err)
+	}
+	watcher, err := pjclient.Watch(metav1.ListOptions{})
```

Comment:
This seems to do an implicit `List`, which is completely unnecessary. I will see if it is possible to have this be some kind of API call barrier so we only get events that happen after the watch starts.

We should also add labels/annotations to the generated jobs to remove the noise from the watch.

Reply:
Yeah, I was aware we will want *some* labels but haven't thought about how they should look like.

Would labelling all entities (not just prowjobs but also rehearsal CMs and the like) with something like `rehearsal: $PRNUMBER` make sense? Would we need some "namespace" for the label (like the `prow.k8s.io` ones have?)?

Reply:
That is what I had in mind, it would be sufficient for this purpose.

As for the namespace, we use `ci.openshift.org` in several places already (ci-operator, secret mirroring, etc.).